bundles/org.openhab.core.config.core/src/main/java/org/openhab/core/config/core/ConfigParser.java

```diff
@@ -127,6 +127,9 @@ private ConfigParser() {
             try {
                 value = valueAs(value, type);
                 if (value == null) {
+                    LOGGER.warn(
+                            "Could not set value for field '{}' because conversion returned null. Check your configuration value.",
+                            fieldName);
```

Comment:
Do you know if the `value` can also be `null` initially... so there is nothing to convert which would then cause unnecessary warnings?

Reply:
I hope @J-N-K can help to answer to this good question.

Reply:
The method exits above (l. 99) when the initial value is `null`. So in this case it's always a non-null value that is passed to `valueAs` and thus `valueAs` only returns with `null` in case of an error.

```
           if (value == null) {
                LOGGER.trace("Skipping field '{}', because config has no entry for {}", fieldName, fieldName);
                continue;
            }
```